assessment/libs/complex.php

```diff
@@ -0,0 +1,636 @@
+    }
+    
+    if (count($num)>1){
+        $ans=$A;
+    }    else{
+            $ans=$A[0];
+        }
+
+    return $ans;
+      
+}
+
+//------------------------------------------Formating to polar form---------------------------------------
+// Function: cx_format2pol(num, [argin="deg" , roundto=3])
+// Returns a string form of a complex number in polar form: r (cos(t) + isin(t)). 
+// 
+// Parameters:
+// num: An array of real and imaginary parts of complex numbers given in square brackets: num = array([Re, Im]).
+//      num array can include more than one complex number. 
+// argin: Optional - Unit for the argument; default is "deg" for degree. For argument in radian, argin = "rad".
+// roundto: Optional - number of decimal places to which modulus should be rounded off; 
+//          default is 3 decimal places. 
+// Returns:
+// A complex number in standard form as a string. If num has more than one complex number, the function returns
+// an array of the polar forms as strings: answer = array("r1 (cos(t1) + isin(t1))", "r2 (cos(t2) + isin(t2))", ...).
+
+function cx_format2pol(array $num, string $argin="rad", int $roundto=3) {
+
+    $A=array();
+    for ($i=0;$i<count($num);$i++){
+        $num1=array($num[$i]);
+        $sq=$num[0][0]**2+$num[0][1]**2;
+        $r= reduceradical($sq);
+        //$r= round(cx_modul($num1),$roundto);
+
+        if ($argin=="rad"){
+                $th0= cx_arg($num1);
+                $th1= round($th0/pi(),$roundto);  
+                $th1=makexxpretty("$th1 pi"); 
+        }
+            else {
+                $th1=round(cx_arg($num1,"deg"),$rounto);
+        }
+        
+        $A[$i]=makexpretty("$r (cos($th1)+isin($th1))");
+    }
+    
+    if (count($num)>1){
+        $ans=$A;
+    }    else{
+            $ans=$A[0];
+        }
+
+    return $ans;
+      
+}
+
+//------------------------------String Formatted Quadratic real and complex roots------------------------
+// Function: cx_prettyquadRoot(a,b,c, [roundto=3])
+// Returns an array of the string of roots of the quadratic equation f(x) = ax^2 + bx + c. Real roots are returned
+// as an array("r1","r2") ordered from the smallest to largest, and the complex roots are returned 
+// as an array("a+bi","a-bi"). This function is suitable for displaying answer (i.e., $showanswer).
+// 
+// Parameters:
+// a: The numerical coefficient of x^2
+// b: The numerical coefficient of x    
+// c: The constant
+// roundto: Optional - number of decimal places to which modulus should be rounded off; 
+//          default is 3 decimal places. 
+//
+// Returns:
+// An array of formatted string of roots (either real or complex) of the quadratic equation.
+
+function cx_prettyquadRoot(float $a, float $b, float $c, int $roundto = 3){
+    $d=$b**2 - 4*$a*$c;
+    
+    if ($d<0){
+        
+            $a2=$a*2;
+            $D=makereducedfraction($b,$a2);
+
+            $re= -$D;
+            $N=reduceradicalfrac(1,-$d,$a2);
```

Comment:
Using this assumes the radical macro library has been loaded.   To make sure, you might want to add at the top of this function:
```
if (!function_exists('reduceradicalfrac')) {
  require_once(__DIR__.'/radicals.php');
}
```
(I think that's right)

Reply:
Thank you, David, for your comments. I am making the changes. 

I was using reducedradical function and I noticed that if I don't wrap the input to the function in a rounding statement, it'd give a wrong answer for sqrt(3)**2, sqrt(13)**2, sqrt(23)**2, sqrt(43)**2, etc. For example,

reducedradical((sqrt(3)**2) = sqrt(2)
reducedradical((sqrt(13)**2) = 2sqrt(3)

Not sure why it's sensitive to "3", but I think rounding the function input value, $in, to 14 or fewer decimal places would fix the issue.